### PR TITLE
solution

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_clock/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,27 +1,53 @@
...
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
+class App extends React.Component {
+  state = {
+    // eslint-disable-next-line react/no-unused-state
+    time: new Date().toLocaleTimeString(),
+    // eslint-disable-next-line react/no-unused-state
+    timerId: undefined,
```
Comment (Anna-Tkachenko) [inline src/App.tsx:10]: Delete this properties from the state of App component
They should be in the state of Clock component

File: src/App.tsx [hunk 2]
```diff
@@ -1,27 +1,53 @@
...
+    // eslint-disable-next-line react/no-unused-state
+    isCounterVisible: true,
+  };
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+  timerId?: NodeJS.Timer ;
```
Comment (Anna-Tkachenko) [inline src/App.tsx:15]: Delete this variable from App component

File: src/App.tsx [hunk 3]
```diff
@@ -1,27 +1,53 @@
...
 
-  // eslint-disable-next-line
-  console.log(timerId);
+  // eslint-disable-next-line react/no-typos
+  componentWillUnMount() {
+    if (this.timerId) {
+      clearInterval(this.timerId);
+    }
+  }
```
Comment (Anna-Tkachenko) [inline src/App.tsx:22]: Same here, you should clearInterval in Clock component, not in App

File: src/Counter.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+export class Counter extends React.Component {
```
Comment (Anna-Tkachenko) [inline src/Counter.tsx:3]: It should be Clock component

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

**Comment by yelnikov-andrii** [comment]
> 



>

**Comment by yelnikov-andrii** [comment]
> Good!

no, it's no good

**Comment by v-vytrykush** [comment]
> > Good!
> 
> no, it's no good

but still, it works)

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**